functions/api/v1/instance.ts

```diff
@@ -1,36 +1,93 @@
+// https://docs.joinmastodon.org/entities/Instance/
+// https://docs.joinmastodon.org/methods/instance/
 import type { Env } from 'wildebeest/backend/src/types/env'
 import { cors } from 'wildebeest/backend/src/utils/cors'
+import * as error from 'wildebeest/backend/src/errors'
 import { DEFAULT_THUMBNAIL } from 'wildebeest/backend/src/config'
 import { getVersion } from 'wildebeest/config/versions'
+import { calculateInstanceStatistics } from 'wildebeest/backend/src/mastodon/instance'
+import { MastodonInstance, InstanceStatistics } from 'wildebeest/backend/src/types/instance'
+import { MastodonAccount } from 'wildebeest/backend/src/types/account'
+import { loadLocalMastodonAccount } from 'wildebeest/backend/src/mastodon/account'
+import { Database, getDatabase } from 'wildebeest/backend/src/database'
+import { getAdmins } from 'wildebeest/backend/src/utils/auth/getAdmins'
+import { emailSymbol } from 'wildebeest/backend/src/activitypub/actors'
 
 export const onRequest: PagesFunction<Env, any> = async ({ env, request }) => {
-	const domain = new URL(request.url).hostname
-	return handleRequest(domain, env)
+	const domain: string = new URL(request.url).hostname
+	const dbOverride: Database = await getDatabase(env)
+	return handleRequest(domain, env, dbOverride)
 }
 
-export async function handleRequest(domain: string, env: Env) {
+export async function handleRequest(domain: string, env: Env, dbOverride?: Database) {
```

Comment:
The reason this is optional, here, is due to a quirk in the playwright tests: the `seo.spec.ts` calls the `/api/v1/instance` endpoint without pulling the `Database` instance binding into the tests' context, so those 20 tests will fail. Meanwhile, the unit tests do not have access to the full `EventContext` which is available to the `PagesFunction` at runtime when deployed to Cloudflare.

Making the `Database` parameter an optional *override* was my workaround for this conundrum. A few lines down, I check to see if the override exists, if it's not, then I `try` to extract the `Database` binding from the `env` parameter (`await getDatabase(env)`). If that throws an error (as playwright e2e testing does), it is caught and treated as if a `Database` binding isn't available.

As an aside, the `Database` binding is necessary to run the query that returns instance-level statistics for `user_count`, `status_count`, and `domain_count`. These specific parameters are necessary for IceCubes app compatibility.

Reply:
sorry I am confused, the playwright tests do rely on your local D1Database (otherwise you wouldn't see any toots, accounts, etc...) so I am not sure I'm understanding the issue 😕 

also playwright is totally unrelated to the application, it practically runs on a separate process and tests your local server, so I am not sure how playwright can affect the app's behaviour in any ways 😕

Reply:
@dario-piotrowicz What you're saying makes sense, and aligns with what I'm seeing. Here's another way to phrase the question: does playwright make the DB binding available via the mocked `EventContext`? If so, then I should be able to run the following code without throwing an exception:
```js
export async function handleRequest(domain: string, env: Env) {
  const db: Database = await getDatabase(env)
}
```

The problem is that running that code throws the following error:
```
GET /explore 200 OK (10.28ms)
TypeError: Invalid state: WritableStream is locked
    at new NodeError (node:internal/errors:371:5)
    at WritableStream.close (node:internal/webstreams/writablestream:210:9)
    at /<REDACTED>/wildebeest/frontend/server/entry.cloudflare-pages.js:20:2366
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.S2 [as next] (/<REDACTED>/wildebeest/frontend/server/entry.cloudflare-pages.js:20:4433)
    at ze3 (/<REDACTED>/wildebeest/frontend/server/entry.cloudflare-pages.js:20:6547) {
  code: 'ERR_INVALID_STATE'
}
```